cmd/notary/tuf.go

```diff
@@ -703,3 +713,33 @@ func getTrustPinning(config *viper.Viper) (trustpinning.TrustPinConfig, error) {
 		Certs:       resultCertMap,
 	}, nil
 }
+
+// authRoundTripper tries to authenticate the requests via multiple HTTP transactions (until first succeed)
+type authRoundTripper struct {
+	trippers []http.RoundTripper
+}
+
+func newAuthRoundTripper(trippers ...http.RoundTripper) http.RoundTripper {
+	return &authRoundTripper{trippers: trippers}
+}
+
+func (a *authRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+
+	var err error
+	var resp *http.Response
+	// Try all run all transactions
+	for _, t := range a.trippers {
+		resp, err = t.RoundTrip(req)
+		// Reject on error
+		if err != nil {
+			return resp, err
+		}
+
+		// Stop when request is authorized/unkown error
+		if resp.StatusCode != http.StatusUnauthorized {
+			return resp, err
+		}
+	}
+
+	return resp, err
```

Comment:
Should be `nil` at line 740 too. And there's no need to pre-declare `err` at 728